modules/pxp-module-puppet

```diff
@@ -221,12 +225,57 @@ def run(params_and_config)
     end
   end
 
-  if !File.exist?(config["puppet_bin"])
-    return make_error_result(DEFAULT_EXITCODE, Errors::NoPuppetBin,
-                             "Puppet executable '#{config["puppet_bin"]}' does not exist")
+  config
+end
+
+def get_action_input(args)
+  action_input = args["input"]
+  action_input["flags"] = action_input["flags"] | ["--onetime", "--no-daemonize", "--verbose"]
+  action_input
+end
+
+def action_run(args)
+  if !args
+    return make_error_result(DEFAULT_EXITCODE, Errors::InvalidJson,
+                             "Invalid json parsed on STDIN. Cannot start run action")
   end
 
-  return start_run(config, params)
+  output_files = args["output_files"]
+  if output_files
+    begin
+      $stdout.reopen(File.open(output_files["stdout"], 'w'))
+      $stderr.reopen(File.open(output_files["stderr"], 'w'))
+    rescue => e
+      print make_error_result(DEFAULT_EXITCODE, Errors::InvalidJson,
+                              "Could not open output files: #{e.message}").to_json
+      exit 5 # this exit code is reserved for problems with opening
+             # of the output_files
+    end
+
+    at_exit do
+      status = if $!.nil?
+        0
+      elsif $!.is_a?(SystemExit)
+        $!.status
+      else
+        1
+      end
+
+      begin
+        File.open(output_files["exitcode"], 'w') do |f|
+          f.print(status)
+        end
+      rescue
+        exit 5 # this exit code is reserved for problems with opening
```

Comment:
It doesn't matter.

If pxp-agent is running, it will consider this a failure because it will get `exitcode == 5`.
If pxp-agent is not running, once it restarts, it will consider this as a failure, since the "exitcode" file is not in the results directory.
This will change anyway when we deal with exception handling; the module has to report its own errors on `stderr` or on the "stderr" file, not on the results object.
Also, consider that this is an edge case; pxp-module-puppet would need to successfully write to "stdout" and "stderr" but fail to write on "exitcode".